hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerRatisServer.java

```diff
@@ -210,23 +209,24 @@ private OMResponse processReply(OMRequest omRequest, RaftClientReply reply)
   }
 
   /**
-   * Parse errorMessage received from the exception and convert to
-   * {@link OzoneManagerProtocolProtos.Status}.
-   * @param errorMessage
-   * @return OzoneManagerProtocolProtos.Status
+   * Convert exception to {@link OzoneManagerProtocolProtos.Status}.
+   * @param cause - Cause from stateMachine exception
+   * @return {@link OzoneManagerProtocolProtos.Status}
    */
-  private OzoneManagerProtocolProtos.Status parseErrorStatus(
-      String errorMessage) {
-    if (errorMessage.contains(STATUS_CODE)) {
-      String errorCode = errorMessage.substring(
-          errorMessage.indexOf(STATUS_CODE) + STATUS_CODE.length());
-      LOG.debug("Parsing error message for error code " +
-          errorCode);
-      return OzoneManagerProtocolProtos.Status.valueOf(errorCode.trim());
+  private OzoneManagerProtocolProtos.Status exceptionToResponseStatus(
+      Throwable cause) {
+    if (cause == null) {
+      // Current Ratis is setting cause, this is an safer side check.
+      LOG.error("StateMachine exception cause is not set");
+      return OzoneManagerProtocolProtos.Status.INTERNAL_ERROR;
+    }
+    if (cause instanceof OMException) {
+      return OzoneManagerProtocolProtos.Status.values()[
+          ((OMException) cause).getResult().ordinal()];
     } else {
+      LOG.error("Unknown error occurs", cause);
       return OzoneManagerProtocolProtos.Status.INTERNAL_ERROR;
```

Comment:
Is there a reason for changing the parsing logic? If I remember correctly, this was added so that even if OMException is wrapped in a RemoteException or a ServiceException, the error code could still be parsed.

Reply:
Previously we used to have some code which used to run in startTransaction for allocateBlock, in that when exception happens, we used to get OMException wrapped in StateMachine Exception and to make parsing easier, we changed errorMessage as STATUS_CODE="<<code>>" Now that logic is removed. So, without current modification, we shall not throw a correct exception status when the cause is OMException.

Reply:
And also now this is called when we receive StateMachineException in RaftClientReply